src/material-experimental/mdc-form-field/form-field.html

```diff
@@ -44,8 +44,11 @@
       </ng-template>
     </div>
 
-    <div class="mat-mdc-form-field-prefix" *ngIf="_prefixChildren.length" #prefixContainer>
-      <ng-content select="[matPrefix]"></ng-content>
+    <div class="mat-mdc-form-field-icon-prefix" *ngIf="_prefixChildren.length" #iconPrefixContainer>
+      <ng-content select="[matPrefix], [matIconPrefix]"></ng-content>
+    </div>
+    <div class="mat-mdc-form-field-text-prefix" *ngIf="_prefixChildren.length" #textPrefixContainer>
```

Comment:
You can avoid one binding with an `ng-container` around both of the containers. E.g.

```
<ng-container *ngIf="_prefixChildren.length">
  <div class="mat-mdc-form-field-icon-prefix" *ngIf="_prefixChildren.length" #iconPrefixContainer></div>
  <div class="mat-mdc-form-field-text-prefix" *ngIf="_prefixChildren.length" #textPrefixContainer></div>
</ng-container>
```

Reply:
I actually tried that and it seemed like the `ng-container` prevented them from being slotted into the appropriate place in the template

Reply:
I think you can use `ngProjectAs` to get around that

Reply:
I still don't think that would help, since I need to project each child as something different

Reply:
Nvm, I commented before I saw the actual template